### ISSUE TITLE
Options do not work on .NET Core in ASP.NET 5 Web template

### ISSUE THREAD

**Issue by rustd** [aspnet/Options#54]
- Create ASP.NET Website template
- Switch to Core CLR
- Run it

The application name is not being set. SiteTitle is being read from config using options model.
In startup we have
```C#
services.Configure<AppSettings>(Configuration.GetSubKey("AppSettings"));
```
In _Layout we have
```C#
@inject IOptions<AppSettings> AppSettings
    <title>@ViewBag.Title - @AppSettings.Options.SiteTitle</title>
```

**Comment by rustd** [comment]
@danroth27 @Eilon @muratg Beta4 bug

**Comment by rustd** [comment]
I think the root cause is GetSubKey is returning null on CoreCLR

**Comment by rustd** [comment]
I think the root cause is GetSubKey is returning null on CoreCLR

**Comment by Eilon** [comment]
Not beta4.

**Comment by muratg** [comment]
We need to fix the Options bug (not sure exactly what the root cause is though.)  We also should update the template not to use Options for storing site title.

**Comment by ppanyukov** [comment]
For what it's worth, I had exactly same problem, and then it somehow gone away. I did this:

- I fiddled around with `project.json`
- this triggered package restore
- I reverted any changes to `project.json`
- another package restore
- after this the problem is gone.

Very strange. Possibly some broken cached version of some package somewhere there?

My `project.json` uses `-beta4` postfixes for everything, as on out-of-the box template for MVC 6 project.

**Comment by regisbsb** [comment]
@rustd the whole file is condition compiled out of DNX core....
It has been fixed in a87122599d8a3187dc227a87f5fe8eb220f2283d . It would be nice have a look at the issues in github before checkin so you can reference then and close then as appropriate. I've spent the whole day trying to debug and help with a pull request to find out that it was fixed in the dev branch by @HaoK. Normally at work I always check the working items before I commit...

**Comment by danroth27** [comment]
Duplicate of #35
